src/java/org/apache/cassandra/io/util/SequentialWriter.java

```diff
@@ -124,6 +125,9 @@ private static FileChannel openChannel(File file)
                     try { channel.close(); }
                     catch (Throwable t2) { t.addSuppressed(t2); }
                 }
+
+                // Invalidate any cache entries that may exist for a previous file with the same name.
+                ChunkCache.removeFileIdFromCache(file);
```

Comment:
in CNDB we use the ChunkCache also for SAI, should we add some specifc handling ? 

cc @pcmanus

Reply:
Moved to the file handle builder to use the supplied cache instance.